apps/accounts/views.py

```diff
@@ -71,7 +71,19 @@ def inner(self, request):
                                            config=settings.FXA_CONFIG)
         except verify.IdentificationError:
             return Response({'error': 'Profile not found.'}, status=401)
-        return fn(self, request, user=find_user(identity), identity=identity)
+        else:
+            identity_user = find_user(identity)
+            if request.user.is_authenticated():
+                if identity_user is not None and identity_user != request.user:
+                    return Response({'error': 'User mismatch.'}, status=422)
+                elif request.user.fxa_id is not None:
+                    return Response(
+                        {'error': 'User already migrated.'}, status=422)
+                else:
+                    user = request.user
```

Comment:
I thought about making it `user = identity_user or request.user` but that could be `None or AnonymousUser()` which would mess things up. I'm not crazy about this but I think it's required :(